[Docs] Add Databricks file store (#4138)
mlrun/mlrun: docs/store/datastore.md

MLRun supports multiple data stores. (More can easily added by extending the `DataStore` class.)
Data stores are referred to using the schema prefix (e.g. `s3://my-bucket/path`). The currently supported schemas and their urls:
* **files** &mdash; local/shared file paths, format: `/file-dir/path/to/file` (Unix) or `C:/dir/file` (Windows)

> * **http, https** &mdash; read data from HTTP sources (read-only), format: `https://host/path/to/file` (Not supported by Spark ingestion)

Reviewer: ```suggestion
* **http, https** &mdash; read data from HTTP sources (read-only), format: `https://host/path/to/file` (Not supported by runtimes "spark" and "remote-spark" )
```

> * **http, https** &mdash; read data from HTTP sources (read-only), format: `https://host/path/to/file` (Not supported by runtimes "spark" and "remote-spark" )

Reviewer: remove space before ")"

* **s3** &mdash; S3 objects (AWS or other endpoints), format: `s3://<bucket>/path/to/file`
* **v3io, v3ios** &mdash; Iguazio v3io data fabric, format: `v3io://[<remote-host>]/<data-container>/path/to/file`
* **az** &mdash; Azure Blob storage, format: `az://<container>/path/to/file`
* **dbfs** &mdash; Databricks storage, format: `dbfs://path/to/file` (Not supported by Spark ingestion)
* **gs, gcs** &mdash; Google Cloud Storage objects, format: `gs://<bucket>/path/to/file`
* **store** &mdash; MLRun versioned artifacts [(see Artifacts)](./artifacts.html), format: `store://artifacts/<project>/<artifact-name>[:tag]`
* **memory** &mdash; in memory data registry for passing data within the same process, format `memory://key`, use `mlrun.datastore.set_in_memory_item(key, value)` 

>    to register in memory data items (byte buffers or DataFrames). (Not supported by Spark ingestion)

Reviewer: ```suggestion
   to register in memory data items (byte buffers or DataFrames). (Not supported by Spark runtimes)
```

>    to register in memory data items (byte buffers or DataFrames). (Not supported by "spark" runtimes)

Reviewer: ```suggestion
   to register in memory data items (byte buffers or DataFrames). (Not supported by all Spark runtimes)
```

* **s3** &mdash; S3 objects (AWS or other endpoints), format: `s3://<bucket>/path/to/file`
* **v3io, v3ios** &mdash; Iguazio v3io data fabric, format: `v3io://[<remote-host>]/<data-container>/path/to/file`
* **az** &mdash; Azure Blob storage, format: `az://<container>/path/to/file`

> * **dbfs** &mdash; Databricks storage, formst: `dbfs://path/to/file`

Author: fixed

and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
contents directly to the query engine.

### Databricks file system
```{Admonition} Note

> Spark ingestion does not support DBFS.

Reviewer: I think this line can be removed - but if not say: Not supported by runtimes "spark" and "remote-spark"

* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this secret or environment variable
may contain the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
contents directly to the query engine.

### Databricks file system
```{Admonition} Note

> DBFS does not support Spark, http, or memory.

Reviewer 2: I find this note very confusing. First of all, it's not that DBFS does not support Spark, it's that Spark as the processing engine does not support `dbfs://` paths. Also, I'm not sure how `http` and `memory` come into play here - @assaf758, can you clarify this statement? What do you refer to?

Reviewer: well, this section of the doc is talking here about datastore, so in that regard I said dbfs datastore does not support spark ingestion engine. But it's semantic - we mean the same :-)
As to http and memory, the discussion was about datastores which have "caveats" of missing support, so I mentioned http and memory as two datastores [showing in our doc](https://docs.mlrun.org/en/latest/store/datastore.html#shared-data-stores) that spark ingestion do not support.
Since it's all the caveats are spark related, i'm fine with moving this to spark section.

Author: @assaf758 
I think it should be in both places. you never know from which side a user starts reading.
Spark - you mean https://docs.mlrun.org/en/latest/runtimes/spark-operator.html, correct?

Reviewer: Yes. notice that there is also a "remote-spark" that does not have a description (the [link](https://docs.mlrun.org/en/latest/concepts/functions-overview.html#kinds-of-functions-runtimes) in the overview section points directly to feature-set ingestion example)

Author: all complete except an overview topic for remote-spark. Noted as AI

* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this secret or environment variable
may contain the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
contents directly to the query engine.

### Databricks file system
* `DATABRICKS_HOST` &mdash; hostname in the format: https://abc-d1e2345f-a6b2.cloud.databricks.com'

> * `DATABRICKS_TOKEN` &mdash; Databricks connection token

Reviewer: connection => access

> * `DATABRICKS_TOKEN` &mdash; Databricks connection token

Reviewer 3: ```suggestion
* `DATABRICKS_TOKEN` &mdash; Databricks connection token.
[Perform Databricks personal access token authentication](https://docs.databricks.com/en/dev-tools/auth.html#databricks-personal-access-token-authentication)
```

Author: fixed
